architecture/jvmfrc/src/main/kotlin/com/lynbrookrobotics/kapuchin/hardware/SPI.kt

```diff
@@ -0,0 +1,18 @@
+package com.lynbrookrobotics.kapuchin.hardware
+
+import java.nio.ByteBuffer
+
+/**
+ * Class that outlines certain methods used by WPILIB's SPI object
+ *
+ * @author Nikash Walia
+ */
+abstract class SPI {
```

Comment:
In my opinion, this abstraction is unnecessary.
The `jvmfrc` module is allowed to interact with WPILIB.
Please use the [WPILIB Serial API](http://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/SerialPort.html) directly.